src/common/secretstore.cpp

```diff
@@ -0,0 +1,209 @@
+{
+    // Two empty secrets are equal but no other secret is equal to the empty
+    // one (even the zero-length one).
+    if ( !m_impl )
+        return !other.m_impl;
+
+    wxSecretValueImpl* const impl = other.m_impl;
+    if ( !impl )
+        return false;
+
+    // Here both objects are not empty, so we can compare them.
+    return m_impl->GetSize() == impl->GetSize() &&
+            memcmp(m_impl->GetData(), impl->GetData(), m_impl->GetSize()) == 0;
+}
+
+// ----------------------------------------------------------------------------
+// Methods forwarded to wxSecretValueImpl
+// ----------------------------------------------------------------------------
+
+size_t wxSecretValue::GetSize() const
+{
+    return m_impl ? m_impl->GetSize() : (size_t)0;
+}
+
+const void *wxSecretValue::GetData() const
+{
+    return m_impl ? m_impl->GetData() : NULL;
+}
+
+#ifndef __WINDOWS__
+
+/* static */
+void wxSecretValue::Wipe(size_t size, void *data)
+{
+    // memset_s() is not present under non-MSW systems anyhow and there doesn't
+    // seem to be any other way to portably ensure that the memory is really
+    // cleared, so just do it in this obvious way.
+    memset(data, 0, size);
+}
+
+#endif // __WINDOWS__
+
+// ============================================================================
+// wxSecretStore implementation
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Memory management
+// ----------------------------------------------------------------------------
+
+wxSecretStore::wxSecretStore(const wxSecretStore& other)
+    : m_impl(other.m_impl)
+{
+    if ( m_impl )
+        m_impl->IncRef();
+}
+
+wxSecretStore::~wxSecretStore()
+{
+    if ( m_impl )
+        m_impl->DecRef();
+}
+
+// ----------------------------------------------------------------------------
+// Methods forwarded to wxSecretStoreImpl
+// ----------------------------------------------------------------------------
+
+bool
+wxSecretStore::Save(const wxString& service,
+                    const wxString& user,
+                    const wxSecretValue& secret)
+{
+    if ( !m_impl )
+        return false;
+
+    if ( !secret.m_impl )
+        return false;
+
+    wxString err;
+    if ( !m_impl->Save(service, user, *secret.m_impl, err) )
+    {
+        wxLogError(_("Saving password for \"%s/%s\" failed: %s"),
+                   service, user, err);
+        return false;
+    }
+
+    return true;
+}
+
+wxSecretValue
+wxSecretStore::Load(const wxString& service, const wxString& user) const
+{
+    if ( !m_impl )
+        return wxSecretValue();
+
+    wxString err;
+    wxSecretValueImpl* const secret = m_impl->Load(service, user, err);
+    if ( !secret )
+    {
+        if ( !err.empty() )
+        {
+            wxLogError(_("Reading password for \"%s/%s\" failed"),
+                       service, user, err);
```

Comment:
You're absolutely right, thanks for noticing this! I simply didn't update the format string after factoring out error generation into the common code (initially, all implementations did it on their own, but I decided that it would be better to centralize it here later).

`Save()` is different from `Load()` and `Delete()` because it always logs an error if it returns false, while the other two may return false (or an empty secret value) if no error really occurred.

I'll add the periods too.